### DISCUSSION TITLE
Third party printing service integrations

### DISCUSSION THREAD

**Discussion by dna113p** [immich-app/immich#11734] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Immich should consider implementing api integrations with some of the popular photo printing services ( Shutterfly, Walgreens, etc...). It would be great to be able to send an album or group of photos to Walgreens to print.

While it may be outside of the scope of what Immich is trying to do - I believe those services also offer affiliate programs which could be another source of income to support Immich development.

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by sirweazel** [comment]
I second this. This is scary... i was typing up this same request 2 hours ago, but didn't get a chance to finish my wording when i got distracted. I was logging on to submit it, when i saw this post! I think with an affiliate link it could be another revenue stream.  It also could match the feature parity with google photos that offers a similar experience.

**Reply by dna113p** [reply]
Haha that's strange - I was surprised I couldn't find any other mentions of the idea when searching.

**Comment by AngelaDMerkel** [comment]
Personally, I very much do not want there to be a financial or corporatised incentive in the app, and I especially don't want there to be integration with commercial services from one large company over another. It's very easy to upload the JPG you want to your printer of choice and be done.

**Comment by apitamy** [comment]
I was going to suggest the same thing. It could be a non-intrusive way for the devs to make some money + be a great feature for users. So a win/win.

**Comment by bo0tzz** [comment]
We have no plans to build integrations with commercial third-party services. This would be a great fit for a community tool using the API.

**Comment by devtechk** [comment]
Anyone who wants build this kind of service?

**Reply by devtechk** [reply]
I'm developing using immich api and lulu api just to  try something cool. If everything is ok I'll share the project :)

**Comment by Isabellae4567** [comment]
I think third-party printing integrations would be a great addition to Immich, especially for users who want an easy way to turn their digital albums into physical prints. Services such as Shutterfly and Walgreens could make the process much more convenient by allowing users to send selected photos directly for printing. I’ve also come across [New York printing services](https://www.vslprint.com/) that offer an interesting range of printing options and could be worth checking out for anyone looking for quality print services. Having more printing partners available could give Immich users greater flexibility while also potentially creating additional revenue opportunities through affiliate partnerships.
